src/components/TextInput.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const size = {
+  default: "300px",
+  medium: "400px",
+  large: "600px"
+}
+
+const TextInputContainer = styled.div`
+  margin: 1em;
+`
+
+const TextInputBox = styled.input.attrs({
+  type: 'text',
+})`
+  background-color: transparent; 
+  width: ${p => p.size ? size[p.size] : size["default"]};
+  padding: 10px;
+  border: 2px solid ${p => p.theme.colors.highlight};
+  border-radius: 7px;
+  color: ${p => p.theme.colors.primary};
+  font-family: 'HK Grotesk';
+  font-size: ${p => p.theme.typography.h3.size};
+  ::placeholder {
+    color: ${p => p.theme.colors.highlight};
+  }
+  :hover {
+    border: 2px solid ${p => p.theme.colors.primary};
+  }
+  :focus {
+    border: 2px solid ${p => p.theme.colors.primary};
+    outline: none;
+  }
+  :disabled {
+    border: 2px solid ${p => p.theme.colors.highlight};
+    opacity: ${p => p.theme.opacity.disabled};
+  }
+  ${p => (p.invalid) &&
+    `border: 2px solid ${p.theme.colors.secondaryWarning};
+    ::placeholder {
+      color: ${p.theme.colors.secondaryWarning};
+    }
+    :hover {
+      border: 2px solid ${p => p.theme.colors.primary};
+    }`
+  }
```

Comment:
idk why but i can't get `:invalid` to work, lmk if anyone has ideas, this is my hacky solution for now

Reply:
I think if you change `${p => p.theme.colors.primary}` to `${p.theme.colors.primary}` it should work. 
It's complaining that `${p => p.theme.colors.primary}` is an invalid property value

before
<img width="1318" alt="Screen Shot 2020-11-09 at 7 56 06 PM" src="https://user-images.githubusercontent.com/38872354/98626006-9e059600-22c5-11eb-8c29-bda23136d347.png">

after
<img width="1347" alt="Screen Shot 2020-11-09 at 7 56 55 PM" src="https://user-images.githubusercontent.com/38872354/98626060-b970a100-22c5-11eb-85b8-27f71a350699.png">

Reply:
omg @panjenny0 wrinkliest brain in the wild west... thank u